packages/next-swc/crates/core/src/next_dynamic.rs

```diff
@@ -236,16 +236,48 @@ impl Fold for NextDynamicPatcher {
                             value: generated,
                         })))];
 
+                    let mut has_ssr_false = false;
+
                     if expr.args.len() == 2 {
                         if let Expr::Object(ObjectLit {
                             props: options_props,
                             ..
                         }) = &*expr.args[1].expr
                         {
+                            for prop in options_props.iter() {
+                                if let Some(KeyValueProp { key, value }) =
+                                    prop.clone().prop().and_then(|p| p.key_value())
```

Comment:
This will clone `prop` even if it's not required.

Reply:
You can do 

```rust
if let Some(KeyValueProp { key, value }) =
    prop.as_ref().and_then(|prop| match prop {
        PropOrSpread::Prop(prop) => match &**prop {
            Prop::KeyValue(kv) => Some(kv),
            _ => None,
        }
        _ => None,
    })
{
}
```

instead

Reply:
With nightly-only feature `box_syntax`, 


You can do 

```rust
if let PropOrSpread::Prop(box Prop::KeyValue(KeyValueProp { key, value })) = prop
{
}
```

but this is nightly-only.

Reply:
You seem to generate these enum accessors for the ast nodes, e. g.

``` rs
pub fn key_value(self) -> Option<KeyValueProp>
// Returns Some if self is of variant KeyValue, and None otherwise.
```

In wonder why there isn't a ref variant to this, e. g. 


``` rs
pub fn key_value_ref(&self) -> Option<&KeyValueProp>
// Returns Some if self is of variant KeyValue, and None otherwise.
```

Reply:
I would like to write this nested `if let` `match` stuff as:

``` rs
if let Some(KeyValueProp { key, value }) =
  try { prop.prop_ref()?.key_value_ref()? }
{
```

or with `Option::map` instead of try blocks^^

Reply:
For ref-based methods, I think it's good to have one. I'll implement it soon.

Reply:
Done. https://github.com/swc-project/swc/pull/3226

It can't be used with current code, because ast type definitions are modified recently.

You can rebase and use methods related to ref after I make a PR to update swc crates, or modify code after merging this PR.
Which do you prefer? 
If you prefer the former, I'll make a upgrade PR asap.

Reply:
I think this can be improved in follow up PRs.